shell/platform/android/io/flutter/embedding/android/FlutterActivityAndFragmentDelegate.java

```diff
@@ -429,6 +417,8 @@ void onStop() {
   void onDestroyView() {
     Log.v(TAG, "onDestroyView()");
     ensureAlive();
+
+    flutterView.detachFromFlutterEngine();
```

Comment:
The big potential drawback of the move to here is that this method [isn't always actually called before the activity is killed](https://developer.android.com/reference/android/app/Activity.html#onDestroy()), so there's a chance that `FlutterView` is never going to see this call in some cases now.

I'm not sure how much of a concern this is because I'm not sure how often this happens on Android, just that it does happen and it's not recommended to actually do critical work on `onDestroy` for that reason. As I understand it this typically happens if Android is about to kill the app for going OOM. The typical "fix" for this is to move actual critical resource releasing into `onPause` instead. I don't _think_ that's really necessary here, but it may be something to revisit if we start seeing a lot of bugs around plugins not getting detached callbacks.

Reply:
Agreed. The reason this was initially done elsewhere is because of the non-guarantee that you mention. But my hope/expectation is that if `onDestroy()` is skipped it's because the app is being evicted, at which point nothing will survive.